packages/java-parser/src/productions/blocks-and-statements.js

```diff
@@ -516,30 +509,17 @@ function defineRules($, t) {
     });
   });
 
-  // https://docs.oracle.com/javase/specs/jls/se16/html/jls-14.html#jls-Resource
+  // https://docs.oracle.com/javase/specs/jls/se21/html/jls-14.html#jls-Resource
   $.RULE("resource", () => {
     $.OR([
       {
-        GATE: $.BACKTRACK($.resourceInit),
-        // Spec Deviation: extracted this alternative to "resourceInit"
-        //                 to enable backtracking.
-        ALT: () => $.SUBRULE($.resourceInit)
+        GATE: () => $.BACKTRACK_LOOKAHEAD($.isLocalVariableDeclaration),
+        ALT: () => $.SUBRULE($.localVariableDeclaration)
       },
       { ALT: () => $.SUBRULE($.variableAccess) }
     ]);
   });
 
-  // Spec Deviation: extracted from "resource"
-  $.RULE("resourceInit", () => {
-    $.MANY(() => {
-      $.SUBRULE($.variableModifier);
-    });
-    $.SUBRULE($.localVariableType);
-    $.CONSUME(t.Identifier);
-    $.CONSUME(t.Equals);
-    $.SUBRULE($.expression);
-  });
```

Comment:
Thanks for the update ! If I may, it would have been even better if you split the commit in two, one for handling the new JDK 22 unnamed variables & patterns feature, and one other for updating this: it is easier to review as it does not mix up two different things :)

Reply:
Good point, I probably should have made this change as the first commit, proving that no changes occurred after making this change, and then have my JDK 22 unnamed variables & patterns feature be the second commit, which would effectively leverage the change from the first. I'll keep this kind of thing in mind for future contributions!